pkg/antctl/raw/supportbundle/command.go

```diff
@@ -0,0 +1,486 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package supportbundle
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/cheggaaa/pb/v3"
+	"github.com/spf13/cobra"
+	"golang.org/x/sync/errgroup"
+	"golang.org/x/time/rate"
+	"gopkg.in/yaml.v2"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog"
+
+	agentapiserver "github.com/vmware-tanzu/antrea/pkg/agent/apiserver"
+	"github.com/vmware-tanzu/antrea/pkg/agent/controller/noderoute"
+	"github.com/vmware-tanzu/antrea/pkg/antctl/runtime"
+	systemv1beta1 "github.com/vmware-tanzu/antrea/pkg/apis/system/v1beta1"
+	controllerapiserver "github.com/vmware-tanzu/antrea/pkg/apiserver"
+	antrea "github.com/vmware-tanzu/antrea/pkg/client/clientset/versioned"
+)
+
+const (
+	barTmpl      pb.ProgressBarTemplate = `{{string . "prefix"}}{{bar . }} {{percent . }} {{rtime . "ETA %s"}}` // Example: 'Prefix[-->______] 20%'
+	requestRate                         = 50
+	requestBurst                        = 100
+)
+
+// Command is the support bundle command implementation.
+var Command *cobra.Command
+
+var option = &struct {
+	dir           string
+	labelSelector string
+}{}
+
+var remoteControllerLongDescription = strings.TrimSpace(`
+Generate support bundles for the cluster, which include: information about each Antrea agent, information about the Antrea controller and general information about the cluster.
+The controller support bundle will always be collected no matter whether the master Node is selected or not.
+`)
+
+var remoteControllerExample = strings.Trim(`
+  Generate support bundles of controller and agent on all Nodes and save them to current working dir
```

Comment:
A question - do we have a way to get only controller bundle?

Reply:
It is possible but a little tricky. We could specify a not exist node name, then no agent bundle would be collected.

Reply:
IMO, we should use a reserved name like "controller" for controller bundle, and for other matching expressions we just include agent bundles of the matched Nodes.
But if it is too late to make the change, we can do a separate PR for that.

Reply:
Got it.